tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -94,38 +95,51 @@ class DeployIosCommand extends Command {
   @override
   Future<void> run() async {
     _throwIfFlavorIsNotSupportForDeployment();
-    await throwIfCodemagiCliToolsAreNotInstalled();
+    await throwIfCodemagicCliToolsAreNotInstalled();
 
     // Is used so that runProcess commands print the command that was run. Right
     // now this can't be done via an argument.
     //
     // This workaround should be addressed in the future.
     isVerbose = true;
 
-    final appStoreConnectConfig = AppStoreConnectConfig.create(
-      argResults!,
-      Platform.environment,
-    );
-
-    const platform = 'IOS';
-    final buildNumber = await getNextBuildNumberFromAppStoreConnect(
-      appStoreConnectConfig: appStoreConnectConfig,
-      platform: platform,
-      // Using the app location as working direcorty because the default
-      // location for the App Store Connect private key is
-      // app/private_keys/AuthKey_{keyIdentifier}.p8.
-      workingDirectory: _repo.sharezoneFlutterApp.path,
-    );
-    await _buildApp(buildNumber: buildNumber);
-    await publishToAppStoreConnect(
-      appStoreConnectConfig: appStoreConnectConfig,
-      stage: argResults![releaseStageOptionName] as String,
-      whatsNew: argResults![whatsNewOptionName] as String?,
-      platform: platform,
-      path: 'build/ios/ipa/*.ipa',
-      repo: _repo,
-      stageToTracks: _iosStageToTracks,
-    );
+    const platform = ApplePlatform.iOS;
+
+    try {
+      await setUpSigning(
+        config: AppleSigningConfig.create(
+          argResults: argResults!,
+          environment: Platform.environment,
+          platform: platform,
+          type: ProvisioningProfileType.iOsAppStore,
+        ),
+      );
+
+      final appStoreConnectConfig = AppStoreConnectConfig.create(
+        argResults!,
+        Platform.environment,
+      );
+
+      final buildNumber = await getNextBuildNumberFromAppStoreConnect(
+        appStoreConnectConfig: appStoreConnectConfig,
+        platform: platform,
+        // Using the app location as working directory because the default
+        // location for the App Store Connect private key is
+        // app/private_keys/AuthKey_{keyIdentifier}.p8.
+        workingDirectory: _repo.sharezoneFlutterApp.path,
+      );
+      await _buildApp(buildNumber: buildNumber);
+      await publishToAppStoreConnect(
+        appStoreConnectConfig: appStoreConnectConfig,
+        stage: argResults![releaseStageOptionName] as String,
+        whatsNew: argResults![whatsNewOptionName] as String?,
+        path: 'build/ios/ipa/*.ipa',
+        repo: _repo,
+        stageToTracks: _iosStageToTracks,
+      );
+    } finally {
+      await keychainUseLogin();
```

Comment:
```suggestion
      // Fixes potential authentication issues after running keychain commands.
      // Only really necessary when running on local machines.
      await keychainUseLogin();
```

I  think we can add this here as well to save ourselves the "hop" to the documentation for this comment in the future.